frontend/components/CellOutput.js

```diff
@@ -282,16 +306,19 @@ const execute_scripttags = async ({ root_node, script_nodes, previous_results_ma
                     }
 
                     const cell = root_node.closest("pluto-cell")
-                    let result = await execute_dynamic_function({
-                        environment: {
-                            this: script_id ? old_result : window,
-                            currentScript: node,
-                            invalidation: invalidation,
-                            getPublishedObject: (id) => cell.getPublishedObject(id),
-                            ...observablehq_for_cells,
-                        },
-                        code: node.innerText,
+                    let { node: new_node, result } = await execute_inside_script_tag_that_replace(node, async () => {
+                        return await execute_dynamic_function({
+                            environment: {
+                                this: script_id ? old_result : window,
+                                currentScript: document.currentScript,
```

Comment:
if we leave this at `node`, it will still work in async things, right?

Reply:
Because right now it would break async stuff?

Reply:
Sorry for the confusion, the "might remove currentScript" in my original message was a suggestion. The `currentScript` variable works exactly as it did before, with and without async.

Reply:
Nice! I guess i should just try it out